beta/nncf/tensorflow/sparsity/rb/functions.py

```diff
@@ -0,0 +1,36 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import tensorflow as tf
+
+from beta.nncf.tensorflow.functions import logit
+
+
+def binary_mask(mask):
+    return tf.round(tf.math.sigmoid(mask))
+
+
+@tf.custom_gradient
+def st_binary_mask(mask):
+    def grad(upstream):
+        return upstream
+    return binary_mask(mask), grad
```

Comment:
Are you bypassing the sigmoid gradient here in addition to the round function?

Reply:
Yes, and it actually contradicts the formula because derivative of sigma not used in calculations:

![\epsilon = \lbrack \sigma(s + \sigma^{-1}(\xi)) > \frac{1}{2} \rbrack, \quad \xi \sim \mathcal{U}(0,1)](https://latex.codecogs.com/png.latex?%5Cdpi%7B120%7D%20%5Cepsilon%20%3D%20%5Clbrack%20%5Csigma%28s%20&plus;%20%5Csigma%5E%7B-1%7D%28%5Cxi%29%29%20%3E%20%5Cfrac%7B1%7D%7B2%7D%20%5Crbrack%2C%20%5Cquad%20%5Cxi%20%5Csim%20%5Cmathcal%7BU%7D%280%2C1%29)

Reply:
Done. @lazarevevgeny, please check it

Reply:
Why do you refer me in the comment? May be you wanted to mention somebody else?